### PR TITLE
Add fake package to azcore

### PR DESCRIPTION
**Pull request by jhendrixMSFT**
This is the supporting infrastructure for the generated SDK fakes.

Partial fix for https://github.com/Azure/azure-sdk-for-go/issues/16613

To be released in the next `azcore` beta.

### CODE REVIEW

**Comment by jhendrixMSFT** [comment]
New public surface area is in the following files.
- sdk/azcore/fake/fake.go
- sdk/runtime/request.go

**Comment by jhendrixMSFT** [comment]
APIView: https://apiview.dev/Assemblies/Review/facd04d16a4e4ee396c29cfcce8d67c9#azcore/fake

**Review by JeffreyRichter — APPROVED** [review]

File: sdk/azcore/runtime/request.go [hunk 1]
```diff
@@ -169,6 +169,9 @@ func SkipBodyDownload(req *policy.Request) {
 	req.SetOperationValue(bodyDownloadPolicyOpValues{Skip: true})
 }
 
+// CtxAPINameKey is used as a context key for adding/retrieving the API name.
+type CtxAPINameKey = shared.CtxAPINameKey
```
[the remark this answers is not in the document]
Comment (jhendrixMSFT) [reply sdk/azcore/runtime/request.go:173]: Jeff and I discussed this offline, we'll leave them separate for now.

**Pull request merged**